Core/include/Acts/EventData/SingleCurvilinearTrackParameters.hpp

```diff
@@ -68,14 +68,21 @@ class SingleCurvilinearTrackParameters
   // this class does not have a custom default constructor and thus should not
   // provide any custom default cstors, dstor, or assignment. see ISOCPP C.20.
 
+  /// Space-time position four-vector.
+  ///
+  /// The surface is owned by the parameters object and thus is independent
+  /// from the geometry context.
+  Vector4D position4() const { return Base::position4(GeometryContext()); }
```

Comment:
I don't think we should default construct the geo context anywhere in our code other than the tests.

Reply:
I agree in principle, but there is no other way here. The curvilinear parameters can be defined anywhere in space and are not restricted to the existing geometry. They own there surface and accessing the position is independent from the geometry context. But, all surface transformations require a context. So it is either that, or to not provide `.position{4}()` at all and require an external context also for the curvilinear parameters.

Reply:
You can reimplement the getter, and I think you should in this case.

Reply:
It still needs to use `Surface::localToGlobal` which always requires a geometry context. Rewritting the getter will end up with exactly the same code as before just with an addditional
```
GeometryContext geoCtx;
```
declaration.

Reply:
Hm, I'm not happy about this, but ok.

Reply:
On second thought, I don't we do (or event should) require `GeometryContext` to be default-constructible outside of our test code, where we plug in an example context.

Reply:
Also, in curvilinear parameters we know the contained surface is a plane surface, so the transform of that is enough. Therefore, just reimplement the `Base::position` here, don't use `localToGlobal` but put a manual transform multiplication in.

Reply:
> Also, in curvilinear parameters we know the contained surface is a plane surface, so the transform of that is enough. Therefore, just reimplement the `Base::position` here, don't use `localToGlobal` but put a manual transform multiplication in.

I know, but that does not help us. One of the main points of the parameters clean up was to reduce the number of variants that we need to support; thus, `CurvilinearTrackParameters` are now just `BoundTrackParameters` with a fixed internal surface. But that also means that we need to support `.referenceSurface()` which must return a `Surface` reference. That means that we must store the `PlaneSurface` and not just the transformation. We could `dynamic_cast<....>` the surface interface to `PlaneSurface` but the `PlaneSurface` does not provide access to the internal transformation **without geometry context** either. As far as I understand it, it can not provide this since the underlying `Surface` could always be based on an underlying detector element and thus **must** be accessed via a geometry context.

So i really do not see how this can be implemented without a default constructed geometry context. I agree with you that we can not assume that the geometry context can be default constructed. My suggestion would then be to remove the geometry context independent position-like accessors altogether for now.

Reply:
True. We could add a getter `defaultTransform` to surface that acts as just a dumb getter for the transform. That would be preferrable to giving meaning to default constructed `GeometryContext` in my opinion.

Otherwise, I think removing the zero-arg version here would be the way to go.

Reply:
The zero-arg versions are removed (as mentioned below).